src/tokenizer/mod.rs

```diff
@@ -1423,4 +1543,34 @@ mod test {
         option_push(&mut s, 'x');
         assert_eq!(s, Some("yx".to_tendril()));
     }
+
+    #[test]
+    fn check_lines() {
+        let opts = TokenizerOpts { exact_errors: false, discard_bom: true, profile: false,
+            initial_state: None, last_start_tag_name: None,
+        };
+        let vector = vec![StrTendril::from("<a>\r"), StrTendril::from("<b>\r"),
+            StrTendril::from("</b>\r"), StrTendril::from("</a>\r")];
```

Comment:
We'll want to use `\n` here instead; all real content will either use `\n` or `\r\n`.